src/main/java/school/hei/haapi/endpoint/rest/controller/FeeController.java

```diff
@@ -16,7 +22,27 @@
   private final FeeMapper feeMapper;
 
   @GetMapping("/students/{studentId}/fees/{feeId}")
-  public Fee getStudentFeeById(@PathVariable String studentId, @PathVariable String feeId) {
-    return feeMapper.toRestFee(feeService.getByUserIdAndId(studentId, feeId));
+  public Fee getFeeByStudentId(@PathVariable String studentId, @PathVariable String feeId) {
+    return feeMapper.toRestFee(feeService.getByStudentIdAndFeeId(studentId, feeId));
+  }
+
+  @PostMapping("/students/{studentId}/fees")
+  public List<Fee> addFee(@PathVariable String studentId, @RequestBody List<CreateFee> toCreate) {
+    return feeService.saveAll(studentId, toCreate
+            .stream()
+            .map(feeMapper::toDomainFee)
+            .collect(toUnmodifiableList()))
+        .stream()
+        .map(feeMapper::toRestFee)
+        .collect(toUnmodifiableList());
+  }
+
+  @GetMapping("/students/{studentId}/fees")
+  public List<Fee> getFeesByStudentId(@PathVariable String studentId) {
+    return feeService
+        .getFeesByStudentId(studentId)
+        .stream()
```

Comment:
42-42 in one line. It's usual to end a ling with `stream()` and continue the stream operation on new line, one line per operation